plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraMetadata.java

```diff
@@ -119,16 +125,31 @@ public CassandraTableHandle getTableHandle(ConnectorSession session, SchemaTable
 
     private static SchemaTableName getTableName(ConnectorTableHandle tableHandle)
     {
-        return ((CassandraTableHandle) tableHandle).getSchemaTableName();
+        return ((CassandraTableHandle) tableHandle).getRequiredNamedRelation().getSchemaTableName();
     }
 
     @Override
     public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         requireNonNull(tableHandle, "tableHandle is null");
+        CassandraTableHandle handle = (CassandraTableHandle) tableHandle;
+        if (handle.getRelationHandle() instanceof CassandraQueryRelationHandle queryRelationHandle) {
+            List<ColumnMetadata> columns = getColumnHandles(queryRelationHandle.getQuery()).stream()
+                    .map(column -> ((CassandraColumnHandle) column).getColumnMetadata())
```

Comment:
Can we split this into two map like `map(CassandraColumnHandle.class::cast).map(CassandraColumnHandle::getColumnMetadata)